lib/Sema/CSGen.cpp

```diff
@@ -3643,11 +3648,12 @@ namespace {
       if (auto closure = dyn_cast<ClosureExpr>(expr)) {
         auto &CS = CG.getConstraintSystem();
         if (closure->hasSingleExpressionBody()) {
-          CG.enterClosure(closure);
+          auto closureType = CG.visitClosureExpr(closure);
+          if (!closureType)
```

Comment:
This should never be `Type()`

Reply:
It can be in `inferClosureType` returns type with an error...